ansible/config_sonic_basedon_testbed.yml

```
@@ -742,16 +742,24 @@
             service: name=chrony state=restarted enabled=true
         when: chrony_service.status.LoadState != "not-found"
 
+      - name: Check if "config bmp" is supported
+        shell: "config -h | grep -q 'bmp'"
+        register: bmp_support_check
+        ignore_errors: true
+        changed_when: false
```

Do we need `changed_when` and `failed_when` if we already ignored error?

---

Thanks @bingwang-ms , the logic is the same just the difference will be the playbook output, without failed_when logs will show as failure even without break the procedure, and without changed_when it will keep task as "OK" of "CHANGED", I think this block is for checking whether bmp feature is supported thus those should not be logged.